File: dsa/python/quickSort.py
def quickSort(arr):
    if len(arr)<=1:
        return arr
    pivot = arr[len(arr)-1]
    print(pivot,"lakdsfljsdf")
    left =[]
    right= []
    for i in range(0,len(arr)-1):
        if(pivot>arr[i]):
            left.append(arr[i])
        else:
            right.append(arr[i])
        
    return [*quickSort(left),pivot,*quickSort(right)]




print(quickSort([3, 6, 8, 10, 1, 2, 1]));

File: dsa/js/sort/quicksort.js
/**
 * Sorts an array using the quicksort algorithm.
 *
 * @param {Array} arr - The array to be sorted.
 * @return {Array} The sorted array.
 */


function quickSort(arr, low = 0, high = arr.length - 1) {
    if (low < high) {
        let pivotIndex = partition(arr, low, high);
        quickSort(arr, low, pivotIndex - 1);
        quickSort(arr, pivotIndex + 1, high);
    }
    return arr;
}

function partition(arr, low, high) {
    let pivot = arr[high];
    let i = low - 1;
    for (let j = low; j < high; j++) {
        if (arr[j] < pivot) {
            i++;
            [arr[i], arr[j]] = [arr[j], arr[i]];
        }
    }
    [arr[i + 1], arr[high]] = [arr[high], arr[i + 1]];
    return i + 1;
}

// Example usage
let arr = [3, 6, 8, 10, 1, 2, 1];
console.log("Sorted array:", quickSort(arr));